Source/Scene/Cesium3DTileBatchTable.js

```diff
@@ -0,0 +1,1519 @@
+            classIds : classIds,
+            classIndexes : classIndexes,
+            parentCounts : parentCounts,
+            parentIndexes : parentIndexes,
+            parentIds : parentIds
+        };
+
+        //>>includeStart('debug', pragmas.debug);
+        validateHierarchy(hierarchy);
+        //>>includeEnd('debug');
+
+        return hierarchy;
+    }
+
+    //>>includeStart('debug', pragmas.debug);
+    var scratchValidateStack = [];
+    function validateHierarchy(hierarchy) {
+        var stack = scratchValidateStack;
+        stack.length = 0;
+
+        var classIds = hierarchy.classIds;
+        var instancesLength = classIds.length;
+
+        for (var i = 0; i < instancesLength; ++i) {
+            validateInstance(hierarchy, i, stack);
+        }
+    }
+
+    function validateInstance(hierarchy, instanceIndex, stack) {
+        var parentCounts = hierarchy.parentCounts;
+        var parentIds = hierarchy.parentIds;
+        var parentIndexes = hierarchy.parentIndexes;
+        var classIds = hierarchy.classIds;
+        var instancesLength = classIds.length;
+
+        if (!defined(parentIds)) {
+            // No need to validate if there are no parents
+            return;
+        }
+
+        if (instanceIndex >= instancesLength) {
+            throw new DeveloperError('Parent index ' + instanceIndex + ' exceeds the total number of instances: ' + instancesLength);
+        }
+        if (stack.indexOf(instanceIndex) > -1) {
+            throw new DeveloperError('Circular dependency detected in the batch table hierarchy.');
+        }
+
+        stack.push(instanceIndex);
+        var parentCount = defined(parentCounts) ? parentCounts[instanceIndex] : 1;
+        var parentIndex = defined(parentCounts) ? parentIndexes[instanceIndex] : instanceIndex;
+        for (var i = 0; i < parentCount; ++i) {
+            var parentId = parentIds[parentIndex + i];
+            // Stop the traversal when the instance has no parent (its parentId equals itself), else continue the traversal.
+            if (parentId !== instanceIndex) {
+                validateInstance(hierarchy, parentId, stack);
+            }
+        }
+        stack.pop(instanceIndex);
+    }
+    //>>includeEnd('debug');
+
+    Cesium3DTileBatchTable.getBinaryProperties = function(featuresLength, json, binary) {
+        var binaryProperties;
+        if (defined(json)) {
+            for (var name in json) {
+                if (json.hasOwnProperty(name)) {
+                    var property = json[name];
+                    var byteOffset = property.byteOffset;
+                    if (defined(byteOffset)) {
+                        // This is a binary property
+                        var componentType = ComponentDatatype.fromName(property.componentType);
+                        var type = property.type;
+                        //>>includeStart('debug', pragmas.debug);
+                        if (!defined(componentType)) {
```

Comment:
I turned these into runtime errors since they would be mistakes in the tile content.